### PR TITLE
Another config matching bug

### PR DESCRIPTION
**Pull request by Juniormunk**
This is quite an odd issue/fix. 

So this is what happened... Photonvision booted with the camera connected and the camera was working...
After a short time the camera stopped working (for some reason maybe static, maybe temp, maybe wiring, idk).
During this time pv showed

Jul 04 06:25:18 BackLeft java[643]: [2024-07-04 06:25:18] [CSCore - PvCSCoreLogger] [ERROR] CS: ERROR 40: ioctl VIDIOC_QBUF failed at UsbCameraImpl.cpp:723: Invalid argument (UsbUtil.cpp:156)
Jul 04 06:25:18 BackLeft java[643]: [2024-07-04 06:25:18] [CSCore - PvCSCoreLogger] [WARN] CS: WARNING 30: BackLeft: could not queue buffer 0 (UsbCameraImpl.cpp:724)

I went over and played with the wire. The camera fully disconnected but it ended up "reconnecting"
When the camera was "reconnected" photonvision detected a "new camera" except this time with no otherpaths (aka no usb path, or by id path).
That resulted in pv creating a new camera configuration for a camera with no otherpaths
Cscore then started to report errors that look like it attempted to connect to the same camera twice 

This fixes it by filtering out USB cameras that have no otherpath on linux.

### CODE REVIEW

**Comment by mcm001** [comment]
Of dmekkfkskskksjfjd course. Thanks V4L. Will dig in tonight

File: photon-core/src/main/java/org/photonvision/vision/processes/VisionSourceManager.java [hunk 1]
```diff
@@ -546,6 +537,13 @@ private List<CameraInfo> filterAllowedDevices(List<CameraInfo> allDevices) {
             } else if (device.name.matches(ignoredCamerasRegex)) {
                 logger.trace("Skipping ignored device: \"" + device.name + "\" at \"" + device.path);
             } else if (device.getIsV4lCsiCamera()) {
+            } else if (device.otherPaths.length == 0
```
Comment (mcm001) [inline photon-core/src/main/java/org/photonvision/vision/processes/VisionSourceManager.java:540]: double else-if - syntax error?

File: photon-core/src/main/java/org/photonvision/vision/processes/VisionSourceManager.java [hunk 2]
```diff
@@ -546,6 +537,13 @@ private List<CameraInfo> filterAllowedDevices(List<CameraInfo> allDevices) {
...
+                    && platform.osType == OSType.LINUX
+                    && device.cameraType == CameraType.UsbCamera) {
+                logger.trace(
+                        "Skipping device with no other paths: \"" + device.name + "\" at \"" + device.path);
+                // If cscore hasnt passed this other paths aka a path by id or a path as in usb port then we
+                // cant guarantee it is a valid camera.
```
Comment (mcm001) [inline photon-core/src/main/java/org/photonvision/vision/processes/VisionSourceManager.java:546]: do we need to make any assertions to check that otherPaths does not contain "by-id" or "by-path", or will the otherPaths length always be empty?

File: photon-targeting/src/main/java/org/photonvision/common/hardware/Platform.java [hunk 3]
```diff
@@ -54,7 +54,7 @@ public enum Platform {
     LINUX_ARM32("Linux ARM32", "linuxarm32", false, OSType.LINUX, false), // ODROID XU4, C1+
     UNKNOWN("Unsupported Platform", "", false, OSType.UNKNOWN, false);
 
-    private enum OSType {
+    public enum OSType {
```
Comment (mcm001) [inline photon-targeting/src/main/java/org/photonvision/common/hardware/Platform.java:57]: Keep this private, prefer to use isLinux() instead.

Comment (mcm001) [reply photon-targeting/src/main/java/org/photonvision/common/hardware/Platform.java:57]: Nvm, would require fixing lots of platform stuff.

File: photon-core/src/main/java/org/photonvision/vision/processes/VisionSourceManager.java (hunk 1 above)
Comment (mcm001) [reply photon-core/src/main/java/org/photonvision/vision/processes/VisionSourceManager.java:540]: skill issue, this is a complete statement, just a confusing one with an empty body. add a comment

File: photon-core/src/main/java/org/photonvision/vision/processes/VisionSourceManager.java (hunk 2 above)
Comment (Juniormunk) [reply photon-core/src/main/java/org/photonvision/vision/processes/VisionSourceManager.java:546]: Good question. This specific problem is hard to recreate. I believe if we get either a by-id or by-path we should be good to assume we can read from the camera.

Comment (mcm001) [reply photon-core/src/main/java/org/photonvision/vision/processes/VisionSourceManager.java:546]: Do we want to merge this as-is, or explicitly check only that there are no by-ids or by-paths?

Comment (Juniormunk) [reply photon-core/src/main/java/org/photonvision/vision/processes/VisionSourceManager.java:546]: im fine doing it as is

**Review by mcm001 — APPROVED** [review]
send it

**Pull request merged**